runtime-interface/src/lib.rs

```diff
@@ -253,3 +258,86 @@ pub trait GearRI {
         println!("Hello from gear runtime interface!!!");
     }
 }
+
+#[cfg(feature = "std")]
+#[cfg(unix)]
+#[test]
+fn test_mprotect_pages_vec() {
+    use gear_core::memory::WasmPageNumber;
+    use libc::{c_void, siginfo_t};
+    use nix::sys::signal;
+
+    const OLD_VALUE: u8 = 99;
+    const NEW_VALUE: u8 = 100;
+
+    extern "C" fn handle_sigsegv(_: i32, info: *mut siginfo_t, _: *mut c_void) {
+        unsafe {
+            let mem = (*info).si_addr() as usize;
+            let ps = page_size::get();
+            let addr = ((mem / ps) * ps) as *mut c_void;
+            if libc::mprotect(addr, ps, libc::PROT_WRITE) != 0 {
+                panic!("Cannot set protection: {}", errno::errno());
+            }
+            for p in 0..ps / PageNumber::size() {
+                *((mem + p * PageNumber::size()) as *mut u8) = NEW_VALUE;
+            }
+            if libc::mprotect(addr, ps, libc::PROT_READ) != 0 {
+                panic!("Cannot set protection: {}", errno::errno());
+            }
+        }
+    }
+
+    let mut v = vec![0u8; 3 * WasmPageNumber::size()];
+    let buff = v.as_mut_ptr() as usize;
+    let page_begin = (((buff + WasmPageNumber::size()) / WasmPageNumber::size())
+        * WasmPageNumber::size()) as u64;
+
+    mprotect_pages_vec(page_begin + 1, &[0, 1, 2, 3], true)
+        .expect_err("Must fail because page_begin + 1 is not aligned addr");
+
+    let pages_to_protect = [0, 1, 2, 3, 16, 17, 18, 19, 20, 21, 22, 23];
+    let pages_unprotected = [
+        4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 24, 25, 26, 27, 28, 29, 30, 31,
+    ];
+
+    // Set [OLD_VALUE] as value for each first byte of gear pages
+    unsafe {
+        for p in pages_to_protect.iter() {
+            let addr = page_begin as usize + *p as usize * PageNumber::size() + 1;
+            *(addr as *mut u8) = OLD_VALUE;
+        }
+        for p in pages_unprotected.iter() {
+            let addr = page_begin as usize + *p as usize * PageNumber::size() + 1;
+            *(addr as *mut u8) = OLD_VALUE;
+        }
```

Comment:
this can be shortened:
```suggestion
        for p in pages_to_protect.iter().chain(&pages_unprotected) {
            let addr = page_begin as usize + *p as usize * PageNumber::size() + 1;
            *(addr as *mut u8) = OLD_VALUE;
        }
```

Reply:
Done